### PR TITLE
Remove OOP related feature options.

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**
This code to make it possible to run features OOP was added around 18 months ago.  Since then there has been no issue with it.  It has also been A/B tested with no stability issues arising, and positive numbers for all latency-oriented scenarios.

This PR simply removes the options for controlling this and makes it so that OOP is on all the time.

Fixes https://github.com/dotnet/roslyn/issues/26076

### CODE REVIEW

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs [hunk 1]
```diff
@@ -11,49 +10,10 @@ internal static class RemoteFeatureOptions
...
-            nameof(RemoteFeatureOptions), nameof(SymbolSearchEnabled), defaultValue: true,
-            storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(SymbolSearchEnabled)));
-
-        public static readonly Option<bool> SymbolFinderEnabled = new Option<bool>(
-            nameof(RemoteFeatureOptions), nameof(SymbolFinderEnabled), defaultValue: true,
-            storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(SymbolFinderEnabled)));
-
         public static readonly Option<bool> DiagnosticsEnabled = new Option<bool>(
             nameof(RemoteFeatureOptions), nameof(DiagnosticsEnabled), defaultValue: true,
             storageLocations: new LocalUserProfileStorageLocation(LocalRegistryPath + nameof(DiagnosticsEnabled)));
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: this option is still around since some tests explicitly set this to false as diagnostics doesn't work OOP in tests (not sure why, but i'm just preserving that behavior).

**Comment by CyrusNajmabadi** [comment]
Tagging @sharwell @jinujoseph @dotnet/roslyn-ide

File: src/EditorFeatures/CSharpTest/AddUsing/AddUsingTests.cs [hunk 2]
```diff
@@ -68,10 +68,7 @@ public partial class AddUsingTests : AbstractAddUsingTests
             Workspace workspace, TestParameters parameters)
         {
             var outOfProcess = (bool)parameters.fixProviderData;
-            workspace.Options = workspace.Options.WithChangedOption(RemoteHostOptions.RemoteHostTest, outOfProcess)
-                                                 .WithChangedOption(RemoteFeatureOptions.OutOfProcessAllowed, outOfProcess)
-                                                 .WithChangedOption(RemoteFeatureOptions.AddImportEnabled, outOfProcess);
-
+            workspace.Options = workspace.Options.WithChangedOption(RemoteHostOptions.RemoteHostTest, outOfProcess);
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/CSharpTest/AddUsing/AddUsingTests.cs]: note: these features fallback to running proc if the oop server doesn't launch.  So it's till good to ahve tests that validate they work properly both in proc and out of proc.

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: @heejaechang Do you know why diagnostics can't run OOP?  The relevant check that forces us to keep this is here:
https://github.com/dotnet/roslyn/blob/f79608d488bcb63336be0d50ced3a87c77108e4c/src/EditorFeatures/TestUtilities/CodeActions/AbstractCodeActionOrUserDiagnosticTest.cs#L375-L379

**Comment by jinujoseph** [comment]
@heejaechang  ,can we rerun the val build again now that the insertion is done

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs (hunk 1 above)
Comment (heejaechang) [reply src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: test doesn't have OOP. you need to explicitly set up mock remote host to simulate OOP in unit test which I do for VS.Net unit test but nowhere else.

File: src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs [hunk 3]
```diff
@@ -105,40 +105,7 @@ internal static class RemoteHostClientExtensions
...
-            this Workspace workspace, Option<bool> featureOption, CancellationToken cancellationToken)
-        {
-            if (!workspace.IsOutOfProcessEnabled(featureOption))
-            {
-                return SpecializedTasks.Default<RemoteHostClient>();
-            }
-
-            return workspace.TryGetRemoteHostClientAsync(cancellationToken);
-        }
+            => workspace.Options.GetOption(featureOption);
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs]: not sure why we still have this option.

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs [hunk 4]
```diff
@@ -11,49 +10,10 @@ internal static class RemoteFeatureOptions
...
 
-        private static ImmutableArray<Option<bool>> AllFeatureOptions { get; } =
-            ImmutableArray.Create(AddImportEnabled, DocumentHighlightingEnabled, NavigateToEnabled, SymbolSearchEnabled, SymbolFinderEnabled);
-
-        public static bool AnyFeatureRunsInProcess(Workspace workspace)
-            => AllFeatureOptions.Any(o => !workspace.IsOutOfProcessEnabled(o));
-
         public static bool ShouldComputeIndex(Workspace workspace)
```
Comment (heejaechang) [inline src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: not sure why we still need this.

**Comment by heejaechang** [comment]
I had similar PR (https://github.com/dotnet/roslyn/pull/29676) and closed  because @sharwell said he has PR with different approach.

File: src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs (hunk 4 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Remote/RemoteFeatureOptions.cs]: This is still necessary because computing the index should only happen in certain workspaces.

File: src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs (hunk 3 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Remote/RemoteHostClientExtensions.cs]: You're right.  Removed.  Thanks! :)

**Comment by CyrusNajmabadi** [comment]
> I had similar PR (#29676) and closed because @sharwell said he has PR with different approach.

Not sure what's up then. AFAICT, OOP is still not enabled by default, and the option for the user is still in the product.

**Comment by sharwell** [comment]
Probably duplicate of #27983 or #27985.

**Comment by CyrusNajmabadi** [comment]
@sharwell there seems to have been no movement there for like 6 months :)

**Comment by sharwell** [comment]
We're going to review the three approaches in design meeting today and move one forward.

**Comment by CyrusNajmabadi** [comment]
> We're going to review the three approaches in design meeting today and move one forward.

Ah.  Good to know.  Feel free to close this out if you prefer another approach.  I didn't realize this was something being worked on :)

**Comment by CyrusNajmabadi** [comment]
Can this be merged in @sharwell @jasonmalinowski ?

**Comment by CyrusNajmabadi** [comment]
Can this be merged in @sharwell @jasonmalinowski ?

**Comment by sharwell** [comment]
@CyrusNajmabadi I believe we are planning to merge #27983 first. If this pull request has other changes we can look into it as a follow-up.

**Comment by CyrusNajmabadi** [comment]
@sharwell @jasonmalinowski now that https://github.com/dotnet/roslyn/pull/27983 has gone in, this is all that's left.  Plz review and merge when convenient for you.  Thanks!

**Comment by CyrusNajmabadi** [comment]
@sharwell @jasonmalinowski now that #27983 has gone in, this is all that's left. Plz review and merge when convenient for you. Thanks!

File: src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs [hunk 5]
```diff
@@ -169,10 +169,10 @@ private static int GetSemicolonLocation(SyntaxNode root, SyntaxNode currentNode,
...
             // If condition is null and caret is in the condition section, as in `for ( ; $$; )`, 
             // we will have bailed earlier due to not being inside supported delimiters
-            => forStatementSyntax.Condition == null ?
-                false :
-                caretPosition > forStatementSyntax.Condition.SpanStart &&
-                caretPosition < forStatementSyntax.Condition.Span.End;
+            => forStatementSyntax.Condition == null
+                ? false
+                : caretPosition > forStatementSyntax.Condition.SpanStart &&
+                  caretPosition < forStatementSyntax.Condition.Span.End;
```
Comment (sharwell) [inline src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: 📝 This change can be reverted (this file is out of scope for the current PR)

Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharp/CompleteStatement/CompleteStatementCommandHandler.cs]: @sharwell i was getting a warning that the formatting here was incorrect.  I like having 0 warnings locally.  So i fixed it up.  Otherwise i think you can just get warning fatigue.

**Comment by CyrusNajmabadi** [comment]
@sharwell Feel free to merge when ready.

**Comment by CyrusNajmabadi** [comment]
ping @sharwell @jinujoseph can we move forward with this?

**Comment by sharwell** [comment]
@CyrusNajmabadi can you update the title of this PR to reflect the new behavior?

**Comment by CyrusNajmabadi** [comment]
@sharwell done.

**Comment by CyrusNajmabadi** [comment]
@sharwell You thumbsupped... but didn't merge :)

**Comment by heejaechang** [comment]
talked with @sharwell  and @jinujoseph and I am going to take this.

**Comment by CyrusNajmabadi** [comment]
Thanks!
